guppylang/diagnostic.py

```diff
@@ -208,10 +208,16 @@ def render_diagnostic(self, diag: Diagnostic) -> None:
         else:
             span = to_span(diag.span)
             level = self.level_str(diag.level)
+            all_spans = [
+                span,
+                *(to_span(child.span) for child in diag.children if child.span),
```

Comment:
nit: took me awhile to figure out what this `*` was about - it does seem like there's one more construction/destruction than I'd expect. It might be less efficient but `[span] + [to_span(child.span) for child in diag.children if child.span]` might be clearer. Up to you!